lim_test.py

```diff
@@ -0,0 +1,259 @@
+        self.assertEqual(self.table_b.mean(), (1+9)/4.)
+        self.assertEqual(self.table_c.mean(), (1+10+9)/(1.+5.+3.))
+        with self.assertRaises(ZeroDivisionError):
+            self.empty_table_a.mean()
+        with self.assertRaises(ZeroDivisionError):
+            self.empty_table_b.mean()
+
+    def test_stddev(self):
+        '''test the stddev() function'''
+        with self.assertRaises(ZeroDivisionError):
+            self.empty_table_a.stddev()
+        #confirms stddev computes correctly for numbers below cutoff for //
+        small_test = lim.LongIntTable(dict((x, 2) for x in range(1, 6)))
+        stddev = ((2*2**2 + 2*1)/5.)**0.5
+        self.assertEqual(small_test.stddev(), round(stddev, 4))
+        small_test = None
+        #confirms stddev computes correctly for numbers above cutoff for //
+        large_test = lim.LongIntTable(dict((x, 10**100) for x in range(1, 6)))
+        self.assertEqual(large_test.stddev(), round(stddev, 4))
+        large_test = None
+        stddev = None
+    def test_merge_update(self):
+        '''test the merge function and the three update functions'''
+
+        answer = [(1, 2), (2, 7), (3, 6)]
+
+        self.table_a.merge(self.table_c.frequency_all())
+        self.assertEqual(answer, self.table_a.frequency_all())
+
+        self.table_a.update_frequency(1, 3)
+        answer[0] = (1, 3)
+        self.assertEqual(answer, self.table_a.frequency_all())
+
+        self.table_a.update_value_ow(1, 3)
+        answer = answer[1:]
+        answer[1] = (3, 3)
+        self.assertEqual(answer, self.table_a.frequency_all())
+
+        self.table_a.update_value_add(3, 2)
+        answer = [(2, 10)]
+        self.assertEqual(answer, self.table_a.frequency_all())
+
+class TestAddRm(unittest.TestCase):
+    '''tests for add and remove of LongIntTable'''
+    def setUp(self):
+        self.basic1 = lim.LongIntTable({0:1})
+        self.basic2 = lim.LongIntTable({0:1})
+        self.table_a = lim.LongIntTable({1:1, 2:2, 3:3})
+        self.table_a1 = lim.LongIntTable({1:1, 2:2, 3:3})
+        self.tuples = [(1, 1), (2, 4)]
+        self.other_tuples = [(1, 1), (3, 1)]
+    def tearDown(self):
+        del self.basic1
+        del self.basic2
+        del self.table_a
+        del self.table_a1
+        del self.tuples
+        del self.other_tuples
+
+    def test_error_raising(self):
+        '''make sure errors are raised and table doesn't mutate'''
+        self.assertRaisesRegexp(ValueError, 'cannot add an empty list',
+                                self.basic1.add, 1, [(1, 0)])
+        self.assertRaisesRegexp(ValueError, 'times must be a positive int',
+                                self.basic1.add, 0, [(1, 1)])
+        self.assertRaisesRegexp(ValueError, 'frequencies may not be negative',
+                                self.basic1.add, 1, [(1, -1)])
+        self.assertEqual(self.basic1.frequency_all(), [(0, 1)])
+    def test_order_zeros(self):
```

Comment:
I'm not sure why you have two examples here. What are you trying to show besides different entry points into the API?